src/hotspot/cpu/x86/x86_32.ad

```diff
@@ -13800,6 +13801,32 @@ instruct cmpFastUnlock(eFlagsReg cr, eRegP object, eAXRegP box, eRegP tmp ) %{
   ins_pipe(pipe_slow);
 %}
 
+instruct cmpFastLockLightweight(eFlagsReg cr, eRegP object, eBXRegP box, eAXRegI eax_reg, eRegP tmp, eRegP thread) %{
+  predicate(LockingMode == LM_LIGHTWEIGHT);
+  match(Set cr (FastLock object box));
+  effect(TEMP eax_reg, TEMP tmp, USE_KILL box, TEMP thread);
```

Comment:
Consider changing USE_KILL box to TEMP box. Same overall considerations (long-term, in a follow-up) as in aarch64.

Reply:
An input cannot be `TEMP` so must change the required input nodes for FastLockNode and FastUnlockNode.

Planned this as a followup RFE, would also change the `eBXRegP` / `rbx_RegP ` constraint to `eRegP` and `rRegP`.

Reply:
Ok, good then. (What a mess. Looking forward to see this cleaned-up)